src/mscorlib/shared/System/Memory.cs

```diff
@@ -163,6 +160,30 @@ private Memory(object obj, int index, int length)
         /// </summary>
         public bool IsEmpty => _length == 0;
 
+        /// <summary>
+        /// For <see cref="Memory{Char}"/>, returns a new instance of string that represents the characters pointed to by the memory.
+        /// Otherwise, returns a <see cref="string"/> with the name of the type and the number of elements.
+        /// </summary>
+        public override string ToString()
+        {
+            if (typeof(T) == typeof(char))
+            {
+                if (_object is string str)
+                {
+                    return str.Substring(_index, _length);
+                }
+                else if (_index < 0)
+                {
+                    return Span.ToString();
+                }
+                else if (_object != null)
+                {
+                    return new string((char[])_object);
```

Comment:
Is this correct?  Don't you also need to factor in _index and _length?

Reply:
Yep, overlooked that case. We need to support sliced memory.